src/dynamics/solver/velocity_constraint_wide.rs

```diff
@@ -148,11 +150,10 @@ impl WVelocityConstraint {
 
                     let r = SimdReal::splat(1.0)
                         / (im1 + im2 + gcross1.gdot(gcross1) + gcross2.gdot(gcross2));
-                    let mut rhs = (vel1 - vel2).dot(&force_dir1);
-                    let use_restitution = rhs.simd_le(-restitution_velocity_threshold);
-                    let rhs_with_restitution = rhs + rhs * restitution;
-                    rhs = rhs_with_restitution.select(use_restitution, rhs);
-                    rhs += dist.simd_max(SimdReal::zero()) * inv_dt;
+                    let projected_velocity = (vel1 - vel2).dot(&force_dir1);
+                    let rhs_resting = projected_velocity + dist.simd_max(SimdReal::zero()) * inv_dt;
+                    let rhs_bouncing = projected_velocity * restitution_plus_1;
+                    let rhs = rhs_bouncing.select(is_bouncy, rhs_resting);
```

Comment:
Same remark as above regarding predictive contacts.
Instead of a `select` we could do:
```rust
let restitution = SimdReal::from(array![|ii| manifold_points[ii][k].restitution; SIMD_WIDTH]);
let is_bouncy = SimdReal::from(array![|ii| manifold_points[ii][k].is_bouncy() as u32 as Real; SIMD_WIDTH]);
let projected_velocity = (vel1 - vel2).dot(&force_dir1);
let rhs = (SimdReal::splat(1.0) + is_bouncy * restitution) * projected_velocity + dist.simd_max(SimdReal::zero()) * inv_dt; 
```